SwiftVoiceQuickstart/ViewController.swift

```diff
@@ -116,111 +113,160 @@ class ViewController: UIViewController, TVONotificationDelegate, TVOCallDelegate
+                self.performStartCallAction(uuid: uuid, handle: handle)
+                return
+            }
+        
+            let alertController = UIAlertController(title: "Voice Quick Start",
+                                                    message: "Microphone permission not granted",
+                                                    preferredStyle: .alert)
+            
+            let continueWithMic = UIAlertAction(title: "Continue without microphone", style: .default) { action in
+                self.performStartCallAction(uuid: uuid, handle: handle)
+            }
+            
+            let goToSettings = UIAlertAction(title: "Settings", style: .default) { action in
+                UIApplication.shared.open(URL(string: UIApplicationOpenSettingsURLString)!,
+                                          options: [UIApplicationOpenURLOptionUniversalLinksOnly: false],
+                                          completionHandler: nil)
+            }
+            
+            let cancel = UIAlertAction(title: "Cancel", style: .cancel) { _ in
+                self.toggleUIState(isEnabled: true, showCallControl: false)
+                self.stopSpin()
             }
+            
+            [continueWithMic, goToSettings, cancel].forEach { alertController.addAction($0) }
+            
+            self.present(alertController, animated: true, completion: nil)
         }
     }
     
     func checkRecordPermission(completion: @escaping (_ permissionGranted: Bool) -> Void) {
-        let permissionStatus: AVAudioSessionRecordPermission = AVAudioSession.sharedInstance().recordPermission()
+        let permissionStatus = AVAudioSession.sharedInstance().recordPermission()
         
         switch permissionStatus {
-        case AVAudioSessionRecordPermission.granted:
+        case .granted:
             // Record permission already granted.
             completion(true)
-            break
-        case AVAudioSessionRecordPermission.denied:
+        case .denied:
             // Record permission denied.
             completion(false)
-            break
-        case AVAudioSessionRecordPermission.undetermined:
+        case .undetermined:
             // Requesting record permission.
             // Optional: pop up app dialog to let the users know if they want to request.
-            AVAudioSession.sharedInstance().requestRecordPermission({ (granted) in
-                completion(granted)
-            })
-            break
+            AVAudioSession.sharedInstance().requestRecordPermission { granted in completion(granted) }
         default:
             completion(false)
-            break
         }
     }
     
     @IBAction func muteSwitchToggled(_ sender: UISwitch) {
         // The sample app supports toggling mute from app UI only on the last connected call.
-        if let call = self.activeCall {
-            call.isMuted = sender.isOn
-        }
+        guard let call = activeCall else { return }
```

Comment:
Normally the convention for unwrapping optional in Swift is to use the same name like `let activeCall = activeCall`.